### PR TITLE
unifty boolean checks

### PR DESCRIPTION
**Pull request by schosterbarak**
unify boolean checks

### CODE REVIEW

File: checkov/terraform/checks/resource/BaseResourceBooleanValueCheck.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+expected_value = True
+
+
+class BaseResourceBooleanValueCheck(BaseResourceCheck):
+    def __init__(self, name, id, categories, supported_resources):
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf):
+        inspected_key = self.get_inspected_key()
+        if dpath.util.search(conf, inspected_key) != {}:
```
Comment (tronxd) [inline checkov/terraform/checks/resource/BaseResourceBooleanValueCheck.py]: can just write 
`if dpath.util.search(conf, inspected_key):`

File: checkov/terraform/checks/resource/BaseResourceBooleanValueCheck.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from abc import abstractmethod
+
+import dpath
+
+from checkov.terraform.models.enums import CheckResult
+
+from checkov.terraform.checks.resource.base_check import BaseResourceCheck
+
+expected_value = True
```
Comment (tronxd) [inline checkov/terraform/checks/resource/BaseResourceBooleanValueCheck.py]: What if in some point we will need to check for `False` or maybe just generalize the class into `BaseResourceValueCheck`? which will check a certain attribute match a certain value, not just `True`
I think this should be refactored as a function of kind:
```
class BaseResourceValueCheck(...):
@abstractmethod
    def get_inspected_key(self):
        raise NotImplemented()
@abstractmethod
    def get_inspected_value(self):
        raise NotImplemented()     
```
and then implement respectively in a specific check.

File: checkov/terraform/checks/resource/BaseResourceBooleanValueCheck.py (hunk 1 above)
Comment (schosterbarak) [reply checkov/terraform/checks/resource/BaseResourceBooleanValueCheck.py]: doesn't it mean "not null"?
not "empty dict" is different. no?

File: checkov/terraform/checks/resource/BaseResourceBooleanValueCheck.py (hunk 2 above)
Comment (schosterbarak) [reply checkov/terraform/checks/resource/BaseResourceBooleanValueCheck.py]: Like 👍  added

File: checkov/terraform/checks/resource/BaseResourceBooleanValueCheck.py (hunk 1 above)
Comment (tronxd) [reply checkov/terraform/checks/resource/BaseResourceBooleanValueCheck.py]: No difference in that case

**Pull request merged**